sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/FailureSafeParser.scala

```diff
@@ -65,8 +66,15 @@ class FailureSafeParser[IN](
         case DropMalformedMode =>
           Iterator.empty
         case FailFastMode =>
-          throw QueryExecutionErrors.malformedRecordsDetectedInRecordParsingError(
-            toResultRow(e.partialResult(), e.record).toString, e)
+          e.getCause match {
+            case exception: SparkRuntimeException if exception.getErrorClass
+              .equals("MALFORMED_RECORD_IN_PARSING.CANNOT_PARSE_JSON_ARRAYS_AS_STRUCTS") =>
```

Comment:
@HyukjinKwon @cloud-fan @srielau Are you ok with such approach?

Reply:
Depends on you, use case class exception to match maybe better way. WDYT?

Reply:
Yep, I think so. Let's add a sub-class of `SparkRuntimeException`, and catch it here.

Reply:
Done